src/cmd/event.js

```diff
@@ -0,0 +1,195 @@
+            }
+        }
+    } catch (err) {
+      console.error('NSFS Event command: exit on error', err.stack || err);
+      process.exit(2);
+    }
+}
+
+async function out_put_events(events, beautify) {
+    if (beautify) {
+        console.table(events);
+    } else {
+        console.log(events);
+    }
+}
+
+async function check_for_event_file() {
+    const path = '/var/log/noobaa_events.log';
+    return await fs_utils.file_exists(path);
+}
+
+async function get_active_events() {
+    const events_info = await get_all_events();
+    let active_events = [];
+    for (const event_info of events_info) {
+        if (event_info) {
+            const event = JSON.parse(event_info.toString());
+            const message = JSON.parse(event.message.toString());
+            message.message_thread = [];
+            const event_time = new Date(event.timestamp);
+            const time_diff = Date.now() - event_time.getTime();
+            if (time_diff < config.EVENT_ACTIVE_INTERVAL) {
+                message.time = new Date(event_time.getTime());
+                const repeated = structure_repeated_events(message, active_events);
+                if (!repeated || active_events.length === 0) {
+                    message.message_thread.push(message.message);
+                    active_events.push(message);
+                }
+            } else {
+                break;
+            }
+        }
+    }
+    active_events = sort_events(active_events);
+    return active_events;
+}
+
+function structure_repeated_events(message, active_events) {
+    let repeated = false;
+    active_events.forEach(function(obj, index) {
+        if (obj.code === message.code) {
+            obj.occurrence = obj.occurrence ? obj.occurrence + 1 : 2;
+            if (obj.message_thread) {
+                obj.message_thread.push(message.message);
+            } else {
+                obj.message_thread = [message.message];
+            }
+            repeated = true;
+        }
+    });
+    return repeated;
+}
+
+async function list_events(time, severity) {
+    const events_info = await get_all_events();
+    const query_time = new Date(time);
+    let result_events = [];
+    events_info.forEach(function(obj, index) {
+        if (obj) {
+            const event = JSON.parse(obj.toString());
+            const message = JSON.parse(event.message.toString());
+            const event_time = new Date(event.timestamp);
+            message.time = event_time;
+            result_events.push(message);
+            if (event_time < query_time) {
+                result_events.pop();
+            }
+            if (severity && message.severity !== severity) {
+                result_events.pop();
+            }
+        }
+    });
+    result_events = await sort_events(result_events);
+    return result_events;
+}
+
+async function get_all_events() {
+    try {
+        const events_logs = await fs.promises.readFile('/var/log/noobaa_events.log');
```

Comment:
a few comments:
1. better to have the path in a constant, since you use it in multiple places. Also it's probably useful to provide a flag to override the default 
2. this file can be rotated by logrotate. shouldn't we get events from all files? what are the requirements?
3. since this file is written by syslog, won't it contain syslog headers that we need to strip before parsing as JSON?

Reply:
@dannyzaken monitoring team don't need a CLI to fetch the event, they will be fetching the event directly from the event file, 
will remove this file, 
Yes, this file can rotate by logrotate, current condition rotate monthly or size in 10M, I am not sure how we can query the previous event log file in this case.
All the valid event properties are under the property `message`, only `timestamp` needs to be fetched from syslog header, and monitoring team is ok with that,
``` {"timestamp":"2023-10-31T06:43:53.127151+00:00","message":"{\"code\":\"noobaa_started\",\"entity_type\":\"NODE\",\"event_type\":\"STATE_CHANGE\",\"message\":\"Noobaa started\",\"scope\":\"NODE\",\"severity\":\"INFO\",\"state\":\"HEALTHY\",\"pid\":650}", "host":"cbf26f4d14d2","severity":"notice","facility":"local2","syslog-tag":"node[650]:","source":"node"} ```